include/range/v3/detail/config.hpp

```diff
@@ -389,6 +389,14 @@ namespace ranges
 #define RANGES_RE_ENABLE_WARNINGS
 #endif
 
+
+#if __has_attribute(no_sanitize)
+#define RANGES_INTENDED_UNSIGNED_OVERFLOW \
```

Comment:
> But at this point we're just arguing semantics

Yes, although I would argue that `UINT_MAX + 1` does not "attempt to create a numeric value that is outside the range that can be represented;" `UINT_MAX + 1` "attempts to create" `0` and succeeds ;)

I apologize for nitpicking at semantics here, but I think framing the problem correctly helps people to better think about how to fix it. The problem is not that unsigned integers can wrap, despite the "unsigned overflow" frame chosen by ubsan. The problem is that a great many C++ users - including the designers of the standard library - have repeatedly chosen to use unsigned types to mean non-negative as if the inability to represent negative numbers is the primary difference between signed and unsigned integers and not the fact that unsigned integers have modulo behavior.

By framing the problem incorrectly, we encourage people to continue using unsigned integers to represent non-negative quantities and to think of "overflow" as a danger they must defend against. We essentially make the problem worse.

So yes, I would like `RANGES_REALLY_UNSIGNED` or `RANGES_INTENTIONALLY_UNSIGNED` or `RANGES_TELL_UBSAN_TO_SHUT_UP` to distinguish functions that use unsigned types on purpose from those that improperly use unsigned to mean non-negative. (I suppose I can live with `RANGES_INTENDED_WRAP_AROUND`.) But if you really like `RANGES_INTENDED_UNSIGNED_OVERFLOW` then just be patient, I don't think @ericniebler is as militant about signed-vs-unsigned as I am. He may tell you to keep it ;)